Datadog SDK OTel Metrics and Logs API Support: Phase 1 (#32610)
DataDog/documentation: content/en/opentelemetry/instrument/api_support/nodejs/metrics.md

`OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE`
: **Description**: Specifies the aggregation `temporality` to use for each instrument kind. <br>
**Accepted values**: `Cumulative`, `Delta`, `LowMemory`. <br>
**Default**: `delta` <br>
**Notes**: This default value is Datadog's recommended configuration and differs from the OpenTelemetry specification's default.

For a complete list of all shared OTLP environment variables, see [OpenTelemetry Environment Variables Interoperability][1].

### Tag and resource attribute precedence

{{% otel-config-precedence %}}

## Migrate from other setups

### Existing OTel setup

If you are already using the OpenTelemetry SDK with a manual OTLP exporter configuration, follow these steps to migrate:

1. Add the Datadog SDK (`dd-trace`) to your project and enable its instrumentation.
2. Remove any code that manually configures the `OTLPMetricsExporter`. The Datadog SDK handles this configuration automatically.
3. Remove the `@opentelemetry/sdk-node` and `@opentelemetry/exporter-otlp` packages from your project's dependencies.
4. Set the `DD_METRICS_OTEL_ENABLED=true` environment variable.

### Existing DogStatsD setup

> If you are currently using the Datadog DogStatsD client and want to migrate to the OpenTelemetry Metrics API, you need to update your instrumentation code. The main difference is that OTel metrics are configured using environment variables rather than code, and you create `Instrument` objects first.

Reviewer: We can link to opentelemetry nodejs onboarding docs here: https://github.com/open-telemetry/opentelemetry-js/blob/4dded45ec67f90998d75e3ddbde9872b6b466183/doc/metrics.md#getting-started

---
title: Node.js OpenTelemetry Metrics API Support
code_lang: metrics
type: multi-code-lang
code_lang_weight: 1
further_reading:
    - link: opentelemetry/correlate/metrics_and_traces
      tag: Documentation
      text: Correlate OpenTelemetry Traces and Metrics
---

## Overview

{{% otel-overview-native lang="Node.js" signal="Metrics" sdk_name="dd-trace-js" %}}

## Prerequisites

> - **Datadog SDK**: dd-trace version [x.y.z] or later.

Reviewer: (will be released in 2 weeks)

```suggestion
- **Datadog SDK**: dd-trace version [5.77.0] or later.
```

> - **OpenTelemetry API**: `@opentelemetry/api` version [x.y.z] or later. (The Datadog SDK provides the implementation for this API).

Reviewer: ```suggestion
- **OpenTelemetry API**: `@opentelemetry/api` version 1.0.0 to 1.10.0. (The Datadog SDK provides the implementation for this API).
```

---
title: Node.js OpenTelemetry Metrics API Support
code_lang: metrics
type: multi-code-lang
code_lang_weight: 1
further_reading:
    - link: opentelemetry/correlate/metrics_and_traces
      tag: Documentation
      text: Correlate OpenTelemetry Traces and Metrics
---

## Overview

{{% otel-overview-native lang="Node.js" signal="Metrics" sdk_name="dd-trace-js" %}}

## Prerequisites

> - **Datadog SDK**: dd-trace version 5.77.0 or later.

Reviewer: ```suggestion
- **Datadog SDK**: dd-trace version 5.60.0 or later.
```
